File: app/src/main/java/org/cnodejs/android/md/display/listener/OnDoubleClickListener.java
package org.cnodejs.android.md.display.listener;

import android.view.View;

public abstract class OnDoubleClickListener implements View.OnClickListener {

    private final long delayTime;
    private long lastClickTime = 0;

    public OnDoubleClickListener(long delayTime) {
        this.delayTime = delayTime;
    }

    @Override
    public final void onClick(View v) {
        long nowClickTime = System.currentTimeMillis();
        if (nowClickTime - lastClickTime > delayTime) {
            lastClickTime = nowClickTime;
        } else {
            onDoubleClick(v);
        }
    }

    public abstract void onDoubleClick(View v);

}

File: app/src/main/java/org/cnodejs/android/md/ui/listener/OnDoubleClickListener.kt
package org.cnodejs.android.md.ui.listener

import android.view.View

abstract class OnDoubleClickListener(private val timeout: Long = 400) : View.OnClickListener {
    private var lastClickTime = 0L

    final override fun onClick(v: View) {
        val nowClickTime = System.currentTimeMillis()
        if (nowClickTime - lastClickTime > timeout) {
            lastClickTime = nowClickTime
        } else {
            onDoubleClick(v)
        }
    }

    abstract fun onDoubleClick(v: View)
}
